### PR TITLE
Show user listings in the market details

### LINKED ISSUE #17: Display all a user's listings of a bastard, even if they're inactive
Currently we display whether a a user has any active or inactive bids on a bastard. We do this to allow a user to cancel bids that are currently inactive (e.g. due to insufficient balance) but that might become active again in the future. This is especially important in light of the recent issues with OpenSea where "re-activated" listings/bids could be exploited.

We need to display the same thing for a user's listings.

### PR DESCRIPTION
**Pull request by christian-reynolds**
Added text to the market details to show if a user has listings.  Also modified CancelListings to allow the active account to modified old listings even if they are no longer the current owner.

### CODE REVIEW

File: src/utils/market.ts [hunk 1]
```diff
@@ -168,6 +193,27 @@ export const getActivity = async (tokenId: number, provider?: providers.Web3Prov
   return activity;
 };
 
+export const getSellOrdersByMaker = async (address: string, provider?: providers.Web3Provider) => {
```
Comment (rkalis) [inline src/utils/market.ts:199]: So this is for the future "account dashboard" feature that displays *all* listings (regardless of token ID)?

File: src/utils/market.ts [hunk 2]
```diff
@@ -15,15 +15,17 @@ export const createRaribleSdk = (provider?: providers.Web3Provider) => {
...
-  const { owner, ownerDisplay } = await getOwner(tokenId, provider);
-  const listings = await getListings(tokenId, provider);
-  const bids = await getBids(tokenId, OrderStatus.ACTIVE, provider);
-  const inactiveBids = await getBids(tokenId, OrderStatus.INACTIVE, provider);
-  const activity = await getActivity(tokenId, provider);
-
-  return { tokenId, owner, ownerDisplay, listings, bids, inactiveBids, activity };
-};
+export const getMarketData =
+  async (tokenId: number, address: string, provider?: providers.Web3Provider): Promise<MarketData> => {
```
Comment (rkalis) [inline src/utils/market.ts]: This kind of line-breaking is unusual for JS/TS code. If a function doesn't fit in one line, it's more idiomatic to reformat this as:

```ts
export const getMarketData = async (
  tokenId: number, 
  address: string, 
  provider?: providers.Web3Provider,
): Promise<MarketData> => {
```

File: src/components/details/BastardDetails.tsx [hunk 3]
```diff
@@ -17,11 +18,12 @@ interface Props {
...
 
 function BastardDetails({ tokenId, settings }: Props) {
-  const { library } = useWeb3React<providers.Web3Provider>();
+  const { account, library } = useWeb3React<providers.Web3Provider>();
   const { result: metadata } = useAsync(getMetadata, [tokenId]);
-  const { result: marketData, execute: updateMarketData } = useAsync(getMarketData, [tokenId, library], {
-    setLoading: (state) => ({ ...state, loading: true }),
-  });
+  const { result: marketData, execute: updateMarketData } =
+    useAsync(getMarketData, [tokenId, account ?? ZERO_ADDRESS, library], {
```
Comment (rkalis) [inline src/components/details/BastardDetails.tsx]: Similar to the other comment, this kind of line-breaking is not common. If it doesn't fit in one line, it's more usual to do something like this:

```ts
const { result: marketData, execute: updateMarketData } = useAsync(
  getMarketData, 
  [tokenId, account ?? ZERO_ADDRESS, library], 
  { setLoading: (state) => ({ ...state, loading: true }) },
 );

**Review by rkalis — APPROVED** [review]
Looks good overall. I left a few comments on formatting, but overall this makes sense. It's probably even good to replace my "bids by owner" code with a similar setup as this one, since this is much more straightforward.

File: src/utils/market.ts (hunk 1 above)
Comment (christian-reynolds) [reply src/utils/market.ts:199]: Correct!  I can remove this code if you want since it's not being used at this time.

**Pull request merged**